docs/src/concepts/postTransaction.md

```diff
@@ -0,0 +1,118 @@
+await arweave.transactions.sign(transaction, key);
+
+// create an uploader that will seed your data to the network
+let uploader = await arweave.transactions.getUploader(transaction);
+
+// run the uploader until it completes the upload.
+while (!uploader.isComplete) {
+  await uploader.uploadChunk();
+}
+```
+
+## Layer 2 Transactions
+Services that increase the scaleability and performance of the underlying Arweave protocol are referred to as Layer 2 solutions. A bundler is one such service that takes multiple individual transactions and bundles them together into a single transaction that is posted to Layer 1. In this way a single transaction at the protocol level can contain tens of thousands of bundled transactions. With only one restriction, bundled transactions cannot transfer AR tokens.
+
+### Initializing Bundlr Network Client
+Layer 2 transactions are posted using `bundlr-network/client`.
+```js
+import Bundlr from '@bundlr-network/client';
+import fs from "fs";
+
+// load the JWK wallet key file from disk
+let key = JSON.parse(fs.readFileSync("walletFile.txt").toString());
+
+// initailze the bundlr SDK
+const bundlr = new Bundlr("http://node1.bundlr.network", "arweave", key);
+```
+
+### Posting a Bundled Transaction
+A difference between posting Layer 1 and bundled Layer 2 transactions is that when using bundlr you must make a deposit on the bundlr node ahead of time. This deposit can be made using AR tokens or a wide variety of other crypto currencies. Another difference is that the bundlr service guarantees your data will arrive on chain.
+```js
+// load the data from disk
+const imageData = fs.readFileSync(`iamges/myImage.png`);
+
+// add a custom tag that tells the gateway how to serve this data in the browser
+const tags = [
+  {name: "Content-Type", value: "image/png"},
+];
+
+// create the bundled transaction and sign it
+const tx = bundlr.createTransaction(imageData, { tags });
+await tx.sign();
+
+// upload the transaction to bundlr for inclusion in a bundle to be posted
+await tx.upload();
+```
+
+## Layer 2 Dispatch
+Another way to post layer 2 transactions is from the browser. While browsers enforce some constraints around the size of data that can be uploaded, browser based wallets are able to post transactions to Layer 2. Arweave browser wallets impllment a `dispatch()` method. An interesting property of this `dispatch()` method is that if the transaction data is < 100KB, the transaction is free.
+```js
+// use arweave-js to create a transaction
+let tx = await arweave.createTransaction({ data:"Hello World!" })
+
+// add some custom tags to the transaction
+tx.addTag('App-Name', 'PublicSquare')
+tx.addTag('Content-Type', 'text/plain')
+tx.addTag('Version', '1.0.1')
+tx.addTag('Type', 'post')
+
+// use the browser wallet to dispatch() the transaction
+let result = await window.arweaveWallet.dispatch(tx);
+
+// log out the transactino id
+console.log(result.id);
+```
+
```

Comment:
Is it worth noting the advantage of layer 2 transactions is speed? Also, I want to try this today, but I think if I fund the bundler node with AR, then I should be able to dispatch more than 100Kb, it does take time for the funding to go through, but if true, it may be worth including here as well. if it works I will change img to support this flow, instead of L1 transactions.

Reply:
Yes, the intent of dispatch() was that if it was unable to post the data for free, it would fall back to an L1 transaction.  I don't think either of the wallet implementations check for a deposit. 
Also, the fall back to L1 transaction was removed from ArConnect, so the behavior is inconsistent right now. 😬